pkg/api/util/clusteragent.go

```diff
@@ -0,0 +1,185 @@
+	clusterAgentAPIRequestHeaders *http.Header
+}
+
+// GetClusterAgentUtil returns or init the ClusterAgentUtil
+func GetClusterAgentUtil() (*ClusterAgentUtil, error) {
+	if globalClusterAgentUtil == nil {
+		globalClusterAgentUtil = &ClusterAgentUtil{}
+		globalClusterAgentUtil.initRetry.SetupRetrier(&retry.Config{
+			Name:          "clusterAgentClient",
+			AttemptMethod: globalClusterAgentUtil.init,
+			Strategy:      retry.RetryCount,
+			RetryCount:    10,
+			RetryDelay:    30 * time.Second,
+		})
+	}
+	err := globalClusterAgentUtil.initRetry.TriggerRetry()
+	if err != nil {
+		log.Debugf("Init error: %s", err)
+		return nil, err
+	}
+	return globalClusterAgentUtil, nil
+}
+
+func (c *ClusterAgentUtil) init() error {
+	var err error
+
+	c.clusterAgentAPIEndpoint, err = getClusterAgentEndpoint()
+	if err != nil {
+		return err
+	}
+
+	authToken := config.Datadog.GetString("cluster_agent_auth_token")
+	if authToken == "" {
+		return fmt.Errorf("empty cluster_agent_auth_token")
+	}
+	if len(authToken) < clusterAgentAuthTokenMinLen {
+		return fmt.Errorf("need at least a length of %d for cluster_agent_auth_token: %d", clusterAgentAuthTokenMinLen, len(authToken))
+	}
+
+	c.clusterAgentAPIRequestHeaders = &http.Header{}
+	c.clusterAgentAPIRequestHeaders.Set(authorizationHeaderKey, fmt.Sprintf("Bearer %s", authToken))
+
+	// TODO remove insecure
+	c.clusterAgentAPIClient = GetClient(false)
+	c.clusterAgentAPIClient.Timeout = 2 * time.Second
+
+	return nil
+}
+
+// getClusterAgentEndpoint provides a validated https endpoint from:
+// 1. configuration key "cluster_agent_url" like https
+// add the https prefix if the scheme isn't specified
+// 2. environment variables associated with "cluster_agent_kubernetes_service_name"
+// ${dcaServiceName}_SERVICE_HOST and ${dcaServiceName}_SERVICE_PORT
+func getClusterAgentEndpoint() (string, error) {
+	const configDcaURL = "cluster_agent_url"
+	const configDcaSvcName = "cluster_agent_kubernetes_service_name"
+
+	dcaURL := config.Datadog.GetString(configDcaURL)
+	if dcaURL != "" {
+		if strings.HasPrefix(dcaURL, "http://") {
+			return "", fmt.Errorf("cannot get cluster agent endpoint, not a https scheme: %s", dcaURL)
+		}
+		if strings.Contains(dcaURL, "://") == false {
+			log.Tracef("adding https scheme to %s: https://%s", dcaURL, dcaURL)
+			dcaURL = fmt.Sprintf("https://%s", dcaURL)
+		}
+		u, err := url.Parse(dcaURL)
+		if err != nil {
+			return "", err
+		}
+		if u.Scheme != "https" {
+			return "", fmt.Errorf("connot get cluster agent endpoint, not a https scheme: %s", u.Scheme)
+		}
+		return u.String(), nil
+	}
+
+	// Construct the URL with the Kubernetes service environment variables
+	// *_SERVICE_HOST and *_SERVICE_PORT
+	dcaSvc := config.Datadog.GetString(configDcaSvcName)
+	if dcaSvc == "" {
+		return "", fmt.Errorf("cannot get a cluster agent endpoint, both %q and %q are empty", configDcaURL, configDcaSvcName)
+	}
+
+	dcaSvc = strings.ToUpper(dcaSvc)
+
+	// host
```

Comment:
I think extracting from the env and properly dealing with the "" is great (also for the //port)